### ISSUE TITLE
invalid-void fails for default value for type argument

### ISSUE THREAD

**Issue by danielrentz** [palantir/tslint#4907]
### Bug Report

- __TSLint version__: 5.19
- __TypeScript version__: 3.7.4
- __Running TSLint via__: grunt-tslint

#### Reproduction using [TSLint Playground](https://palantir.github.io/tslint-playground)

Cannot reproduce in playground...

#### TypeScript code being linted

```ts
export type Callback<T, U = void> = (value: T) => U;
```

with `tslint.json` configuration:

```json
"rules": {
  "invalid-void": true
}
```

#### Actual behavior

The "invalid-void" rule triggers.

#### Expected behavior

It is possible to use "void" as default value for the type argument.

**Comment by danielrentz** [comment]
Similar situation: Calling a generic function with explicit type argument:

```
function newPromise<T>(): Promise<T> { ... }
const promise = newPromise<void>();
```

**Comment by JoshuaKGoldberg** [comment]
Confirmed this also happens in `6.0.0-beta1`. Accepting PRs for the next week or so, per #4534!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
